### PR TITLE
Record connection latency metric

### PR DESCRIPTION
**Pull request by maayanf24**
Related issue #678

Signed-off-by: Maayan Friedman <maafried@redhat.com>

### CODE REVIEW

File: pkg/cableengine/syncer/syncer.go [hunk 1]
```diff
@@ -209,6 +210,9 @@ func (i *GatewaySyncer) generateGatewayObject() *v1.Gateway {
 				latencyInfo := i.healthCheck.GetLatencyInfo(&connection.Endpoint)
 				if latencyInfo != nil {
 					connection.Latency = latencyInfo.Spec
+					cable.RecordConnectionLatency(localEndpoint.Spec.CableName, &localEndpoint.Spec, &connection.Endpoint,
+						float64(latencyInfo.Spec.AverageRTT/uint64(time.Second)))
```
Comment (mangelajo) [inline pkg/cableengine/syncer/syncer.go:214]: In the case of prometheus, I think it'd be better to expose the LastRTT, in that way prometheus can be queried for average over specific times with it's query language.

**Review by mangelajo — COMMENTED** [review]
AverageRTT vs LastRTT

**Review by dfarrell07 — APPROVED** [review]

File: pkg/cableengine/syncer/syncer.go [hunk 2]
```diff
@@ -209,6 +210,9 @@ func (i *GatewaySyncer) generateGatewayObject() *v1.Gateway {
 				latencyInfo := i.healthCheck.GetLatencyInfo(&connection.Endpoint)
 				if latencyInfo != nil {
 					connection.Latency = latencyInfo.Spec
+					cable.RecordConnectionLatency(localEndpoint.Spec.CableName, &localEndpoint.Spec, &connection.Endpoint,
+						float64(latencyInfo.Spec.LastRTT/uint64(time.Second)))
```
Comment (tpantelis) [inline pkg/cableengine/syncer/syncer.go:214]: Shouldn't we only record this if connected (ie `connection.Status == v1. Connected`)?

**Comment by skitt** [comment]
This will need to be squashed when merged.

**Review by mangelajo — APPROVED** [review]

**Review by mangelajo — APPROVED** [review]

**Pull request merged**